### PR TITLE
Search pagination. Empty state.

### PR DESCRIPTION
**Pull request by pink-bunny**

### CODE REVIEW

File: src/pages/Dashboard/Search/container.js [hunk 1]
```diff
@@ -15,8 +15,8 @@ const SearchForm = withFormik({
 
   validationSchema: SearchFormSchema,
 
-  handleSubmit: (values, { props, setErrors, setSubmitting }) => {
-    props.searchMovies(values, setErrors, setSubmitting);
+  handleSubmit: (values, { props }) => {
+    props.searchMovies(undefined, values);
```
Comment (plzen) [inline src/pages/Dashboard/Search/container.js]: Use values as first argument, optional arguments better to add to the end

File: state/dashboard/actions.js [hunk 2]
```diff
@@ -5,22 +5,25 @@ export const fetchTrendingMovies = (page = 1) => ({
...
   type: types.SEARCH_MOVIES,
-  search,
-  page,
-  setErrors,
-  setSubmitting
+  searchQuery: search,
+  page
+});
+
+export const setSearchMoviesQuery = (searchQuery) => ({
```
Comment (plzen) [inline state/dashboard/actions.js]: You can reuse searchMovies and save query to state

File: state/dashboard/operations/searchMovies.js [hunk 3]
```diff
@@ -3,34 +3,37 @@ import { normalize, schema } from 'normalizr';
...
-    const {
-      search,
-      setErrors,
-      setSubmitting,
-      page
-    } = action;
+  async process({ getState, httpClient, action }, dispatch, done) {
+    const { searchQuery, page } = action;
     const currentPage = page;
+    const query = searchQuery || getState().dashboard.searchQuery;
```
Comment (plzen) [inline state/dashboard/operations/searchMovies.js]: Use selector to get values from state

File: state/dashboard/operations/searchMovies.js [hunk 4]
```diff
@@ -3,34 +3,37 @@ import { normalize, schema } from 'normalizr';
...
     try {
-      const { data } = await httpClient.get(`search/movie?query=${search}&page=${page}`);
+      const { data } = await httpClient.get(`search/movie?query=${query}&page=${page}`);
       // normalize data
       const movie = new schema.Entity('movies');
       const moviesSchema = { results: [movie] };
       const normalizedData = normalize(data, moviesSchema);
-      const { movies } = normalizedData.entities;
-      const ids = normalizedData.result.results;
+      const movies = normalizedData.entities.movies || {};
```
Comment (plzen) [inline state/dashboard/operations/searchMovies.js]: Is it required to use empty object as fallback?

File: src/pages/Dashboard/Search/container.js (hunk 1 above)
Comment (pink-bunny) [reply src/pages/Dashboard/Search/container.js]: Done

File: state/dashboard/actions.js (hunk 2 above)
Comment (pink-bunny) [reply state/dashboard/actions.js]: Removed

File: state/dashboard/operations/searchMovies.js (hunk 3 above)
Comment (pink-bunny) [reply state/dashboard/operations/searchMovies.js]: Now we don't need to take searchQuery from the state.
searchMovies action always has a query.

File: state/dashboard/operations/searchMovies.js (hunk 4 above)
Comment (pink-bunny) [reply state/dashboard/operations/searchMovies.js]: Yes it is. Replaced condition to actions.

File: src/pages/Dashboard/container.js [hunk 5]
```diff
@@ -17,17 +17,22 @@ import {
 } from '../../../state/dashboard/actions';
 
 class DashboardPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.checkSearchState = this.checkSearchState.bind(this);
```
Comment (plzen) [inline src/pages/Dashboard/container.js]: To avoid binding you can use arrow functions

Comment (pink-bunny) [reply src/pages/Dashboard/container.js]: Done.

**Pull request merged**